46 register component api call to backend

Magatsuken opened the pull request:
API call to backend
- Post request upon valid registration form
- Post request sent to correct endpoint
- console.log successful login upon successful login with status code 200
- Generates the correct error message upon failure

Magatsuken committed:
Trim unused code

Magatsuken committed:
Add error messages for register component

Magatsuken committed:
Fix to error handling

Magatsuken committed:
Add API call to backend and error handling

Magatsuken committed:
Add connection to backend test

ndohertyjr commented on client/src/Login/Login.js:
```diff
@@ -43,9 +40,7 @@ export default function Login() {
             .then(function (response) {
```
use anonymous like this : .then(response = () => { rest of the function }

Magatsuken replied:
Done

ndohertyjr commented on client/src/Login/Login.js:
```diff
@@ -114,7 +109,9 @@ export default function Login() {
-                console.log("Unhandled error code");
+                setColor(ToastConstants.color.error);
+                setMessage("Unhandled error code");
+                setShow(true);
```
should be setShowToast

Magatsuken replied:
Done

ndohertyjr commented on client/src/Register/Register.js:
```diff
@@ -1,15 +1,19 @@
 
 // FIXME: Review formating of extra columns
 export default function Register() {
+    const [show, setShow] = useState(false);
```
setShowToast

Magatsuken replied:
Done

ndohertyjr commented on client/src/Register/Register.js:
```diff
@@ -20,6 +24,34 @@ export default function Register() {
+            })
+        };
+
+        fetch('https://localhost:5001/api/auth/register', requestOptions)
```
look into configs and string interpolation for domain changes

Magatsuken replied:
Done

Magatsuken committed:
Update API endpoint variable

Magatsuken committed:
Update both components

- Update show/setShow useState to be more descriptive
- Use env variable for backend API
- Change function to anonymous function for consistency

Magatsuken committed:
Fix spelling errors
